### ISSUE TITLE
Equivalent of `--dir` option for tests

### ISSUE THREAD

**Issue by michalkucharczyk** [paritytech/zombienet#578]
Currently `--dir` option is only supported for `spawn` command.

To implement automated [warp sync test](https://github.com/paritytech/substrate/issues/11471), the predefined database must be passed to nodes spawned for test.

Could you extend `test` command with support for `--dir` option?

It would also be very useful if `test --dir` combination was supported in `podman` and `kubernetes` providers.

**Comment by pepoviola** [comment]
Also, use `relative` paths for `--dir` doesn't work as expected. We should also check that.
Thanks!!

**Comment by michalkucharczyk** [comment]
I am now analyzing how to use the predefined snapshot in CI pipelines.
Seems that test is executed by: [`run-test-env-manager.sh`](https://github.com/paritytech/zombienet/blob/master//scripts/ci/run-test-env-manager.sh) script, which in turn is called from [`zombienet.yml`](https://github.com/paritytech/polkadot/blob/master/scripts/ci/gitlab/pipeline/zombienet.yml) pipeline file.

This means that we should extend `run-test-env-manager.sh` with ability to use predefined directory and pass it to `zombienet` binary, which probably is not the best approach (as the downloading the snapshot, paths and arguments would spread across many files).

The other solution I can imagine is to embed the `database snapshot` (or the `directory snapshot`) into the zombienet configuration file and allow the `zombienet` binary to download it to the directory and _connect_ to the node. This could be alternatice to `--dir` option.

What are your thoughts on this?

**Comment by pepoviola** [comment]
I think this is a good option, but also we can introduce `configurable inits containers` as we had before and that allow us to more flexibility. Basically those are containers with the volume mounted that run before the main container. 
So, you can write an small shell script to download the needed files and extract in the needed path.

Thanks!

**Comment by michalkucharczyk** [comment]
Do `configurable inits containers` allow for seamless switch between providers (e.g. for local execution)? Where the location of `snapshot` would be stored? Could you point me to some example?

My main concern is that having an extra script that is run in CI, and also must be run locally with some extra parameters may make running tests locally difficult. End user trying to execute tests locally should not be concerned with location of files to be downloaded and the location where the path should be extracted and the parameters that should be passed to `zombienet` executable. Extra step may make test running more difficult and potentially error-prone.

**Comment by pepoviola** [comment]
Yes, the init containers will add more flexibility **but** move the complexity to the developer that need to write the script. 
Back to the original idea of having the snapshot set **by** node (node_group)

```
dir_snapshot="s3://bucket/path-to-test/0001-test-warp-sync/snapshots/bob.tgz"
```

Couple of questions:

- compression format? (gzip?)
- This file will only contains the `db` files for the chain of the node.

Thanks!

**Comment by michalkucharczyk** [comment]
> * compression format? (gzip?)
Don't have opinion here. `gzip` is fine for me.

> * This file will only contains the `db` files for the chain of the node.
correct. I would assume that archive contains the following tree:
```
.
`-- data
    `-- chains
        `-- local_testnet
            `-- db
                `-- full
                    |-- 000xxx.log
                    |-- 000xxx.sst
                    |-- CURRENT
                    |-- db_version
                    |-- IDENTITY
                    |-- LOCK
                    |-- LOG
                    |-- MANIFEST-000009
                    |-- OPTIONS-000035
                    `-- OPTIONS-000037
```

**Comment by pepoviola** [comment]
> chains

Can the archive contains the `tree` from the chain (e.g `local_testnet`), since the `data` directory is the mounted volume in both podman/k8s.

What do you think @michalkucharczyk ?

Thanks!

**Comment by michalkucharczyk** [comment]
> > chains
> 
> Can the archive contains the `tree` from the chain (e.g `local_testnet`), since the `data` directory is the mounted volume in both podman/k8s.

ok for me.

**Comment by michalkucharczyk** [comment]
> Can the archive contains the `tree` from the chain (e.g `local_testnet`), since the `data` directory is the mounted volume in both podman/k8s.

For the record:
`tar -C db-snapshot/alice/data/chains/  -czf chains.tgz local_testnet`

```
$ tar tzf chains.tgz | head
local_testnet/
local_testnet/db/
local_testnet/db/full/
...
local_testnet/db/full/000469.log
```

**Comment by michalkucharczyk** [comment]
Closed by #604
